src/main/java/net/dv8tion/jda/internal/entities/emoji/EmojiImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.entities.emoji;
+
+import net.dv8tion.jda.api.entities.emoji.CustomEmoji;
+import net.dv8tion.jda.api.entities.emoji.Emoji;
+import net.dv8tion.jda.api.entities.emoji.UnicodeEmoji;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.utils.EncodingUtil;
+import net.dv8tion.jda.internal.utils.Helpers;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+public class EmojiImpl implements Emoji, CustomEmoji, UnicodeEmoji
```

Comment:
Implementing this like this means that users cannot confidently typecheck an `Emoji` to know if it is `Custom` or `Unicode`, so I'm not sure I'm on board for this.

I was expecting separate `Impl` classes for Custom and Unicode

Reply:
Mixing the types in this impl makes it more confusing for new contributors or maintainers too imo.

Reply:
I prefer this